src/rational/mat_q/get.rs

```diff
@@ -15,6 +15,38 @@ use crate::{error::MathError, rational::Q};
 use flint_sys::{fmpq::fmpq_set, fmpq_mat::fmpq_mat_entry};
 use std::fmt::Display;
 
+impl GetNumRows for MatQ {
+    /// Returns the number of rows of the matrix as a [`i64`].
```

Comment:
Would it be better to return it as a u64? (I am not sure)

Advantages:
- It represents the data better
- Using unsigned integers later might allow to avoid a check that a value is positive when accessing an index.

Disadvantages:
- not what flint uses internally

Reply:
We had this decision in the challenge, flint uses i64 and we didn't want to convert all the time (short version)